lib/library.cpp

```diff
@@ -1208,13 +1208,13 @@ bool Library::matchArguments(const Token *ftok, const std::string &functionName)
         return (callargs == 0);
     int args = 0;
     int firstOptionalArg = -1;
-    for (std::map<int, ArgumentChecks>::const_iterator it2 = it->second.argumentChecks.cbegin(); it2 != it->second.argumentChecks.cend(); ++it2) {
-        if (it2->first > args)
-            args = it2->first;
-        if (it2->second.optional && (firstOptionalArg == -1 || firstOptionalArg > it2->first))
-            firstOptionalArg = it2->first;
+    for (const auto & argCheck : it->second.argumentChecks) {
+        if (argCheck.first > args)
```

Comment:
In my humble opinion the typeinfo is useful when accessing first and second. Perhaps replace the `auto` in your code with `std::pair<int, ArgumentChecks>`. Or perhaps explicitly write the members in the loop body.

    int argnr = argCheck.first;
    const ArgumentChecks& argumentChecks = argCheck.second;

Reply:
Fair point, I'll send another PR.